hpp-fcl.tf

```
@@ -0,0 +1,19 @@
+locals {
+  hpp_fcl_team = [
+    "jcarpent",
+    "jmirabel",
+    "wxmerkt",
+    "nim65s",
+  ]
+  hpp_fcl_repositories = [
+    "hpp-fcl-release",
```

```suggestion
    "hpp_fcl-release",
```
I'll suggest that we use `_` in the team, repository, and terraform file names to match the ROS package naming conventions.

---

For ROS1 and thus the package name, there have been exceptions for dashes for packages that are third-party/CMake packages that do not generate ROS messages (i.e. explicitly hpp-fcl and dynamic-graph). Their `package.xml` for instance uses a dash rather than an underscore as a result. There was some previous discussion [here](https://github.com/ros/rosdistro/pull/25488#discussion_r461023884).

I am happy to modify it for an underscore for the team and repository. Would it still be okay to have the package be released with a dash as that is its name? (Hence we wouldn't have to have two package.xml variants either)

---

> I am happy to modify it for an underscore for the team and repository. Would it still be okay to have the package be released with a dash as that is its name? (Hence we wouldn't have to have two package.xml variants either)

I will use the opportunity to supply just a bit of friction. If we don't take this opportunity to change to a REP-144 compliant package name when would we? Once all ROS 1 distributions are in end-of-support? Never? There is an ever-present Sword of Damocles for a package in ROS which doesn't adhere to the naming conventions as future tools are not guaranteed nor required to support them. Juggling two package.xml files to maintain the status quo where it has already taken hold will expire in 2025 but having a package name that's out-of-spec will last until a hard break is made to correct it. As I interpret REP-144 (and perhaps the REP could do with a clarifying edit if there are other interpretations favored) the name used in package.xml files for third-party packages _must_ also comply with the package naming requirements. what the package calls itself internally or names its headers is 

Since the package name isn't directly part of this PR and you're alright with the elements which are. I'll proceed with making those changes and merging. I am not putting my foot down about the package name but I am hesitant to allow legacy exceptions to apply indefinitely especially when ROS 2 is such a golden opportunity to correct some of them.

---

I fully understand - the decision here for either course of action isn't mine to take. For reducing complexity, it would be an easier route vs vendoring as with `yaml_cpp_vendor` or patching the release with extra CMake config files (*). I'll also cc @nim65s in whose purview a different naming would fall.

(*) I've tried a `condition` on the `name` tag to switch the package name between ROS1/ROS2 but that doesn't work since at least catkin-tools checks that there is only one `name` tag in the `package.xml` - hence we'd need to patch it on release and provide extra CMake configs.

---

Sorry guys, COVID is currently preventing me for thinking straight. I'll trust your decisions here.

---

> I've tried a condition on the name tag to switch the package name between ROS1/ROS2 but that doesn't work since at least catkin-tools checks that there is only one name tag in the package.xml - hence we'd need to patch it on release and provide extra CMake configs.

I believe condition attributes are only supported / standardized on the various dependency tags and on the export/build_type tag. I don't think there's a single-source solution to using different package names between ROS 1 and ROS 2.

---

> Sorry guys, COVID is currently preventing me from thinking straight. I'll trust your decisions here.

That's awful to hear. I wish you rest and a swift recovery 💌